agent/engine/docker_task_engine.go

```diff
@@ -1835,6 +1841,19 @@ func (engine *DockerTaskEngine) inspectContainer(task *apitask.Task, container *
 }
 
 func (engine *DockerTaskEngine) stopContainer(task *apitask.Task, container *apicontainer.Container) dockerapi.DockerContainerMetadata {
+	// Before attempting to stop any container, send drain signal for Appnet Agent to start draining connections
+	// (if not already in progress).
+	if task.IsServiceConnectEnabled() && !task.IsServiceConnectConnectionDraining() {
+		if err := appnetDrainInboundConnections(task.GetServiceConnectRuntimeConfig()); err != nil {
```

Comment:
We probably want to have some max number of retries otherwise task can potentially get stuck in stopping.

Reply:
This doesn't prevent a container from stopping?

Reply:
Oh sorry I misread, thought we'd just return. Are we okay with the best-effort approach (no retries)? I wonder if there can be some type transient errors from AppNet Agent that could potentially go away with a retry.

Reply:
I could add some retries internally in the client. Although. as-is, the next container that stops will retry, but it's probably better to do the retries internally in the client.

Reply:
Added retries inside the function. I think that should suffice for out purposes.